src/WeekCalendar/Cell.tsx

```diff
@@ -0,0 +1,62 @@
+import { Event } from '../types';
+import CalendarEvent from './CalendarEvent';
+import { useRef, useState, useEffect } from 'react';
+import styles from './weekCalendar.module.css';
+import {
+  getEventHeight,
+  getEventWidth,
+  getMarginLeft,
+  getMarginTop,
+} from '../utils';
+
+interface CellProps {
+  cellDate: Date;
+  onCellClick: (date: Date) => void;
+  events: Event[];
+}
+
+const Cell = ({ cellDate, onCellClick, events }: CellProps) => {
+  const cellRef = useRef<HTMLDivElement>(null);
+  const [cellDimensions, setCellDimensions] = useState({ width: 0, height: 0 });
+
+  useEffect(() => {
+    if (cellRef.current) {
+      setCellDimensions({
+        width: cellRef.current.offsetWidth - 10,
+        height: cellRef.current.offsetHeight,
+      });
+    }
+  }, [events, cellDate]);
```

Comment:
do you really need to trigger useEffect when events or cellDate change?
Also, is condition in if correct?